### ISSUE TITLE
Try to get a not exposed resource returns a 401 instead of 404

### ISSUE THREAD

**Issue by tifabien** [api-platform/core#5831]
**API Platform version(s) affected**: 3.1.14

**Description**  

I think this is related to the https://github.com/api-platform/core/pull/5722 fix. 

I have configured a resource which is not accessible through REST (only accessible with graphql) and this resource is protected with an bearer token. 

Before the 3.1.14 version, when I try to access this resource without token I've got a 404 as a returned status. 
Now with the 3.1.14+ this 404 is replaced by a 401.
 
**How to reproduce**  

Create an API resource with the following attributes: 
```
#[ApiResource(
    operations: [],
    security: 'is_granted("ROLE_READ")',
    ...
)]
```

When you try to access a resource item without token, a 401 is now returned and not a 404 anymore.

**Possible Solution**  

**Additional Context**

**Comment by soyuka** [comment]
Yes a resource must have at least one Get operation just declare one with a 404 status.
